airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/state_converters/datetime_stream_state_converter.py

```diff
@@ -113,23 +122,36 @@ def convert_to_sequential_state(self, cursor_field: CursorField, stream_state: M
         """
         if self.is_state_message_compatible(stream_state):
             legacy_state = stream_state.get("legacy", {})
-            if slices := stream_state.pop("slices", None):
-                latest_complete_time = self._get_latest_complete_time(slices)
-                if latest_complete_time:
-                    legacy_state.update({cursor_field.cursor_field_key: self.output_format(latest_complete_time)})
+            latest_complete_time = self._get_latest_complete_time(stream_state["start"], stream_state.get("slices", []))
+            if latest_complete_time is not None:
+                legacy_state.update({cursor_field.cursor_field_key: self.output_format(latest_complete_time)})
             return legacy_state or {}
         else:
             return stream_state
 
-    def _get_latest_complete_time(self, slices: List[MutableMapping[str, Any]]) -> Optional[datetime]:
+    def _get_latest_complete_time(self, start: datetime, slices: List[MutableMapping[str, Any]]) -> Optional[datetime]:
         """
         Get the latest time before which all records have been processed.
         """
         if slices:
-            first_interval = self.merge_intervals(slices)[0][self.END_KEY]
-            return first_interval
+            merged_intervals = self.merge_intervals(slices)
+            first_interval = merged_intervals[0]
+            if start < first_interval[self.START_KEY]:
```

Comment:
Given that `start` is defined as `self.parse_timestamp(state["start"]) if "start" in state else self.zero_value` [here](https://github.com/airbytehq/airbyte/pull/34131/files#diff-44999ccc78e8a64a5a79f8352f8aa2a45d8dde6ecaa92d3dff0cc773de104197R47), I think will never update the state.

Assuming it is the first incremental sync, `start = self.zero_value` which is `0001-01-01T00:00:00.000Z` so we would have the following scenario:
* Start incremental concurrent sync with `config["start"] = 2023-01-01`
* Async generation of partitions `{"start": "2023-01-01", "end": "2023-12-31"}`
* Async processing of `{"start": "2023-01-01", "end": "2023-12-31"}`. Closing this partition. At that point, state is
```
        {
            "state_type": ConcurrencyCompatibleStateType.date_range.value,
            "metadata": { … },
            "start": "0001-01-01T00:00:00.000Z"
            "slices": [
                {"start": "2023-01-01", "end": "2023-12-31"}
            ]
        }
```
* Merging intervals (not relevant here)
* `start < first_interval[self.START_KEY] == True` so we return `0001-01-01T00:00:00.000Z` as latest completed time

Outcome
* Next sync starts from "0001-01-01T00:00:00.000Z"

Expected result
* Next sync starts from "2023-12-31"

It feels like the value of `start` should be initialized as the lower boundaries of all the slices. Based on @girarda's comment yesterday, the easiest wait to have that might be to ask the developer to provide it.

Reply:
Right, this should be taking the start from the config into account too. Will make that update!